farmRe.py

```diff
@@ -0,0 +1,360 @@
+import statistics as stats
+import functions as functions
+
+class Farm:
+    
+    
+    def __init__(self, _farmerName, _cuerdas, _treeType, _initialAgeOfTrees, _sowDensity = functions.hectaresToCuerdas(1000)): # use self to declare namespace
+        self.farmerName = _farmerName
+
+        self.inheretTreeProperties(_treeType)
+
+        self.totalCuerdas = _cuerdas
+        self.sowDensity = _sowDensity # sow density in trees/hectare
+        self.totalTrees = int(round((_cuerdas * _sowDensity), 0)) # round to nearest and convert to int because we can't have poritions of trees
+            # and it needs to be an int to iterate through
+
+            # adjust the initial age of trees to be a round number
+
+        # initialize variables for pruning so if/else can deal with objects
+        self.pruneYear = False
+        self.pruneCount = 0    
+        
+        self.initialAgeOfTrees= self.convertToRoundInt(_initialAgeOfTrees)
+
+        
+        # initialize trees in a list so if we plant another
+        # set of differently aged trees we can add (indicies must match)
+        self.trees = [self.setTrees()] # quantity of trees per set
+        self.ageOfTrees = [self.initialAgeOfTrees] # age of trees
+        
+        
+        self.treesInProd = self.getTreesInProd()
+        # average age of trees is not so easy now because we need to weight it
+        #self.averageAgeOfTrees = stats.mean(self.ageOfTrees)
+
+        
+        self.harvestPerTree = self.cuerdaHarvestCap / _sowDensity # pull initial sow density because the other will change
+        
+        # if plants are added or if others die
+        self.totalHarvest = 0 # units, in this case pounds
+
+
+
+        
+        
+        
+    def convertToRoundInt(self, number): 
+        """
+        
+        A function to assure that specific numbers in the class are rounded and/or converted to type: int
```

Comment:
ahhhh we shouldn't be doing this I think. 
once we case types in our functions, let python enforce this. raise an error if something is a string or float, don't convert it for them under the hood.